### PR TITLE
feat(analyze): add an option to allow generating empty baseline

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by staabm** [comment]
Whats the motivation of this PR?

In case you are interessted in a 0 exit code, you can use
`phpstan ... --generate-baseline || true`

**Comment by Lctrs** [comment]
This will not generate an empty baseline file. And by empty, I mean with this content : 

```
parameters:
	ignoreErrors: []
```

**Comment by Lctrs** [comment]
See https://github.com/phpstan/phpstan-src/pull/751#issuecomment-973018169

File: src/Command/AnalyseCommand.php [hunk 1]
```diff
@@ -55,6 +55,7 @@ protected function configure(): void
 				new InputOption('autoload-file', 'a', InputOption::VALUE_REQUIRED, 'Project\'s additional autoload file path'),
 				new InputOption('error-format', null, InputOption::VALUE_REQUIRED, 'Format in which to print the result of the analysis', null),
 				new InputOption('generate-baseline', null, InputOption::VALUE_OPTIONAL, 'Path to a file where the baseline should be saved', false),
+				new InputOption('allow-empty-baseline', null, InputOption::VALUE_NONE, 'Do not error out when the generated baseline is empty'),
```
Comment (ondrejmirtes) [inline src/Command/AnalyseCommand.php:58]: This is missing a default

File: src/Command/AnalyseCommand.php [hunk 2]
```diff
@@ -102,6 +103,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 			$generateBaselineFile = 'phpstan-baseline.neon';
 		}
 
+		$allowEmptyBaseline = (bool) $input->getOption('allow-empty-baseline');
```
Comment (ondrejmirtes) [inline src/Command/AnalyseCommand.php:106]: This set to `true` should result in an error if `$generateBaselineFile` is null.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Command/AnalyseCommand.php (hunk 1 above)
Comment (Lctrs) [reply src/Command/AnalyseCommand.php:58]: You can't set a default on an input option which has no value.

Comment (ondrejmirtes) [reply src/Command/AnalyseCommand.php:58]: Oh right, haven't realized that 👍

File: src/Command/AnalyseCommand.php (hunk 2 above)
Comment (Lctrs) [reply src/Command/AnalyseCommand.php:106]: Done. I'm not very satisfied with the error message, if you have a better idea...

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**